### PR TITLE
Update fixed point library

### LINKED ISSUE #78: Use struct instead of uint256 for fixed point library
- Move `uintToInt()` and `intToUint()` to another library
- Use struct to represent fixed point variables

### PR DESCRIPTION
**Pull request by haythem96**
# Update fixed point library

## High Level Description

- Update fixed point library to use struct to represent scaled number instead of a simple int variable
- Moved `uintToInt()` and `intToUint()` to a new `SignedConverter` library because they are not related to FixedPoint lib.

Closes #78 .

### Code

- [X] Unit test 100% coverage
- [X] Does your code follow the naming and code documentation guidelines?

### Documentation

- [X] Is your code up to date with the spec? 
- [X] Have you added your tests to the testing doc?

### CODE REVIEW

File: test/signedConverter.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    it('Should convert from signed integer to unsigned integer', async () => {
+      const int = new BigNumber(-3)
+      const expectedUint = new BigNumber(3)
+
+      assert.equal(
+        (await lib.testFromInt(int)).toNumber(),
+        expectedUint.toNumber(),
+        'conversion from int to uint mismatch',
+      )
+    })
```
Comment (aparnakr) [inline test/signedConverter.test.ts:42]: can we maybe have the conversion from signed to unsigned error for -ve numbers and have a separate absolute value function? That way makes it super clear and makes sure we don't mess up.

File: test/signedConverter.test.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+      assert.equal(
+        (await lib.testFromUint(uint)).toNumber(),
+        expectedInt.toNumber(),
+        'conversion from uint to int mismatch',
+      )
+    })
+
+    it('It should revert converting an unsigned integer greater than uint256(-1) to signed integer', async () => {
+      const uint = new BigNumber(2).multipliedBy(1e256).minus(1)
```
Comment (antoncoding) [inline test/signedConverter.test.ts:28]: Isn't this `2*10^256 -1`? and uint256(-1) should be `2^256 - 1` 🤔

**Review by antoncoding — APPROVED** [review]

**Comment by antoncoding** [comment]
> Doesn't this also affect the way margin calculator is implemented?

yes

**Comment by antoncoding** [comment]
Merging now. Opened another issue for abs() #85

**Pull request merged**